### PR TITLE
#48 - Improve code-coverage & add GH check action

### LINKED ISSUE #48: Improve code-coverage & add GH check action
## Background
Improve or add code coverage support to be able measure code quality.

## Feature
New ability to measure current code coverage as one of QA metrics.
Add GH action to check changed file coverage.

### PR DESCRIPTION
**Pull request by miroslavpojer**
- Added support for code coverage GitHub action check as part of PR.
- Used Absa fork of JaCoCo solution - with scala method filtering.

Closes #48

### CODE REVIEW

File: .github/workflows/jacoco_check.yml [hunk 1]
```diff
@@ -0,0 +1,125 @@
+#
+# Copyright 2022 ABSA Group Limited
```
Comment (miroslavpojer) [inline .github/workflows/jacoco_check.yml:2]:
```suggestion
# Copyright 2023 ABSA Group Limited
```

File: .github/workflows/jacoco_check.yml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    strategy:
+      matrix:
+        include:
+          - scala_2_11: 2.11.12
+            scala_2_12: 2.12.12
+            spark_24: 2.4.7
+            spark_32: 3.2.3
+            spark_33: 3.2.1
+            overall: 0.0
+            changed: 80.0
```
[the remark this answers is not in the document]
Comment (miroslavpojer) [reply .github/workflows/jacoco_check.yml:36]: Yes, it is single instance. I am using this solution to work with multiple output dir from cross build.

File: .github/workflows/jacoco_check.yml [hunk 3]
```diff
@@ -0,0 +1,125 @@
+#
+# Copyright 2023 ABSA Group Limited
```
Comment (Zejnilovic) [inline .github/workflows/jacoco_check.yml:2]:
```suggestion
# Copyright 2021 ABSA Group Limited
```

File: project/plugins.sbt [hunk 4]
```diff
@@ -31,5 +31,23 @@ addSbtPlugin("com.eed3si9n"      % "sbt-assembly"      % "1.1.0")
 addSbtPlugin("de.heikoseeberger" % "sbt-header"        % "5.6.0")
 addSbtPlugin("com.github.sbt"    % "sbt-release"       % "1.1.0")
 addSbtPlugin("com.github.sbt"    % "sbt-pgp"           % "2.1.2")
-addSbtPlugin("com.github.sbt"    % "sbt-jacoco"        % "3.4.0")
 addSbtPlugin("com.github.sbt"    % "sbt-ci-release"    % "1.5.10")
+
```
Comment (Zejnilovic) [inline project/plugins.sbt:35]: I know you didn't do it, but please remove one of the licence headers in this file. For some reason it is double.

File: .github/workflows/jacoco_check.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (Zejnilovic) [reply .github/workflows/jacoco_check.yml:36]: If this is not a matrix that needs list X list of possibilities, please don't use the matrix strategy.

If you want variable use https://docs.github.com/en/actions/learn-github-actions/variables

File: project/plugins.sbt (hunk 4 above)
Comment (miroslavpojer) [reply project/plugins.sbt:22]: Fixed in commit c172f508a4148078977b1734bd3ce32d2e81a329.

File: .github/workflows/jacoco_check.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (miroslavpojer) [reply .github/workflows/jacoco_check.yml]: Matrix removed in commit fb157b3d509b3473797ff617e312e1434b321497.

**Review by Zejnilovic — APPROVED** [review]

**Pull request merged**